CRM/Utils/SQL/TempTable.php

```diff
@@ -317,6 +317,16 @@ public function setId($id) {
    * @return $this
    */
   public function setMemory($value = TRUE) {
+    if ($value) {
+      $tablename = 'civicrm_testing_memory_' . $this->id;
```

Comment:
@demeritcowboy the fact that we are remembering `$id`, but not remembering the outcome of this check vexes me. As written, calling this function 10 times would create and drop the same table 10 times, when once ought to be sufficient (and would close fewer transactions).

For that matter, this check is something that really only ever needs to run once. It could run as part of `CRM_Core_Invoke::rebuildMenuAndCaches()` and get cached as a setting.

Reply:
Ok that's a good idea. I ran into some problems with cache getting cleared so the problem kept coming back, but I worked around it. Let's see if this passes: https://github.com/civicrm/civicrm-core/pull/31660